centreon/src/Centreon/Application/Controller/Configuration/MonitoringServerController.php

```diff
@@ -216,9 +216,18 @@ private function execute(callable $callable): void
          */
         $user = $this->getUser();
         try {
-            if (! $user->isAdmin() && ! $user->hasRole(Contact::ROLE_GENERATE_CONFIGURATION)) {
+            if (! $user->hasRole(Contact::ROLE_GENERATE_CONFIGURATION)) {
                 throw new AccessDeniedException('Insufficient rights (required: ROLE_GENERATE_CONFIGURATION)');
             }
+
+            if (
+                ! $user->hasTopologyRole(Contact::ROLE_CONFIGURATION_MONITORING_SERVER_READ)
+                && ! $user->hasTopologyRole(Contact::ROLE_CONFIGURATION_MONITORING_SERVER_READ_WRITE)
+            ) {
+                throw new AccessDeniedException(
+                    'Insufficient rights (required: ROLE_CONFIGURATION_MONITORING_SERVER_READ or ROLE_CONFIGURATION_MONITORING_SERVER_READ_WRITE)'
+                );
+            }
```

Comment:
Should be moved for each use case (GenerateConfiguration and ReloadConfiguration)

Reply:
Done